tests/nn/test_moe.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+import torch
+import torch.nn as nn
+
+from modalities.nn.moe import MoEExpertGLU, MoEExperts, MoEFFN, MoEFFNConfig, MoERouter
+
+
+def test_moe_router_produces_expected_shapes(
+    model_input: torch.Tensor,
+    batch_size: int,
+    seq_length: int,
+    moe_top_k: int,
+    moe_router: MoERouter,
+):
+    top_weights, top_experts = moe_router.forward(model_input)
+    assert top_weights.shape == (batch_size * seq_length, moe_top_k)
+    assert top_experts.shape == (batch_size * seq_length, moe_top_k)
+
+
+def test_moe_router_jitter_does_not_change_shape(model_input: torch.Tensor, moe_router: MoERouter):
+    jittered_model_input = moe_router._jitter(model_input)
+    assert jittered_model_input.shape == model_input.shape
+
+
+def test_moe_expert_produces_expected_shape(
+    model_input: torch.Tensor, batch_size: int, seq_length: int, hidden_size: int, moe_num_experts: int
+):
+    ffn_hidden_size = 128
+    act_fn = nn.ReLU
+    model = MoEExpertGLU(hidden_size, ffn_hidden_size, moe_num_experts, act_fn)
+    expert_idx = 2
+    output = model.forward(model_input, expert_idx)
+    assert output.shape == (batch_size, seq_length, hidden_size)
```

Comment:
The output has the correct shape. Is it intended that the output is filled with nan values? This could be because of the random test configs maybe?